retriever/test/retriever.test.js

```diff
@@ -410,34 +397,74 @@ describe('retriever.fetch', () => {
     const rootCid =
       'baga6ea4seaqaleibb6ud4xeemuzzpsyhl6cxlsymsnfco4cdjka5uzajo2x4ipa'
     const owner = Object.keys(OWNER_TO_RETRIEVAL_URL_MAPPING)[0]
-
-    await env.DB.batch([
-      env.DB.prepare(
-        `
-        INSERT INTO indexer_proof_sets (set_id, owner)
-        VALUES (?, ?)
-      `,
-      ).bind(proofSetId, owner),
-
-      env.DB.prepare(
-        `
-        INSERT INTO indexer_roots (root_id, set_id, root_cid)
-        VALUES (?, ?, ?)
-      `,
-      ).bind(rootId, proofSetId, rootCid),
-      env.DB.prepare(
-        `
-        INSERT INTO indexer_proof_set_rails (proof_set_id, rail_id, payer, payee, with_cdn)
-        VALUES (?, ?, ?, ?, ?)
-      `,
-      ).bind(proofSetId, railId, defaultClientAddress, owner, false),
-    ])
+    await withDbEntries(env, {
+      owner,
+      rootCid,
+      proofSetId,
+      railId,
+      withCDN: false,
+      rootId,
+    })
 
     const req = withRequest(defaultClientAddress, rootCid, 'GET')
     const res = await worker.fetch(req, env)
 
     assert.strictEqual(res.status, 402)
   })
+  it('should fall back to the database and fetch the URL when providerAddress is not found in the mapping', async () => {
+    const providerAddress = '0x2A06D234246eD18b6C91de8349fF34C22C7268e9'
+    const clientAddress = '0x1234567890abcdef1234567890abcdef12345608'
+    const rootCid = 'bagaTest'
+    const body = 'file content'
+
+    // Ensure the mapping does not have this address
+    delete OWNER_TO_RETRIEVAL_URL_MAPPING[providerAddress]
```

Comment:
This can lead to unexpected side effects, when the address actually should be part of the mapping. Safer:

```suggestion
    // Ensure the mapping does not have this address
    assert(!OWNER_TO_RETRIEVAL_URL_MAPPING[providerAddress])
```